mne/report.py

```diff
@@ -1391,23 +1490,52 @@ def save(self, fname=None, open_browser=True, overwrite=False):
             if answer.lower() == 'y':
                 overwrite = True
 
+        _, ext = op.splitext(fname)
+        is_hdf5 = (ext.lower() == '.h5' or ext.lower() == '.hdf5')
+
         if overwrite or not op.isfile(fname):
             logger.info('Saving report to location %s' % fname)
-            fobj = codecs.open(fname, 'w', 'utf-8')
-            fobj.write(_fix_global_ids(u''.join(self.html)))
-            fobj.close()
-
-            # remove header, TOC and footer to allow more saves
-            self.html.pop(0)
-            self.html.pop(0)
-            self.html.pop()
 
-        if open_browser:
+            if is_hdf5:
+                write_hdf5(fname, self.__getstate__(), overwrite=overwrite,
+                           title='mnepython')
+            else:
+                try:
+                    self._render_toc()
+
+                    with warnings.catch_warnings(record=True):
+                        warnings.simplefilter('ignore')
+                        html = footer_template.substitute(
+                            date=time.strftime("%B %d, %Y"),
+                            current_year=time.strftime("%Y"))
+                    self.html.append(html)
+
+                    # Write HTML
+                    with codecs.open(fname, 'w', 'utf-8') as fobj:
+                        fobj.write(_fix_global_ids(u''.join(self.html)))
+                finally:
+                    # remove header, TOC and footer to allow more saves
+                    self.html.pop(0)
+                    self.html.pop(0)
+                    self.html.pop()
```

Comment:
We usually avoid using try in the code. Can you avoid it?

Reply:
In this case, I think the `try`/`finally` construct makes a lot of sense. Even if something breaks during the rendering of the report, we must always revert the addition of the header+footer, otherwise any re-attempt at saving will result in a distorted HTML file. Note that we don't catch any exceptions. Why is this construct usually avoided in the code?

Reply:
ok I get it now.

Maybe add some text to explain this.

btw did you experience this? if so in what context? how come do you predict some issues in this try block?